functions.cpp

```diff
@@ -96,4 +96,40 @@ std::string absolute_path(){
     }
 }
 
+std::string strip_accents(std::string str) {
+    const static std::vector<std::pair<std::string, std::string> > vec_str {
+        
+        {"à", "a"},  {"À", "a"},  {"â", "a"},  {"Â", "a"},  {"ä", "a"},  {"Ä", "a"},  {"Á", "a"},  {"á", "a"},
+        {"Ã", "a"},  {"ã", "a"},  {"Å", "a"},  {"å", "a"},  {"Ǎ", "a"},  {"ǎ", "a"},  {"Ą", "a"},  {"ą", "a"},
+        {"Ă", "a"},  {"ă", "a"},  {"Ā", "a"},  {"ā", "a"},  {"Ç", "c"},  {"ç", "c"},  {"Ć", "c"},  {"ć", "c"},
+        {"Ĉ", "c"},  {"ĉ", "c"},  {"Č", "c"},  {"č", "c"},  {"Ď", "d"},  {"đ", "d"},  {"Đ", "d"},  {"ď", "d"},
+        {"ð", "d"},  {"é", "e"},  {"É", "e"},  {"è", "e"},  {"È", "e"},  {"ê", "e"},  {"Ê", "e"},  {"ë", "e"},
+        {"Ë", "e"},  {"Ě", "e"},  {"ě", "e"},  {"Ę", "e"},  {"ę", "e"},  {"Ė", "e"},  {"ė", "e"},  {"Ē", "e"},
+        {"ē", "e"},  {"Ĝ", "g"},  {"ĝ", "g"},  {"Ģ", "g"},  {"ģ", "g"},  {"Ğ", "g"},  {"ğ", "g"},  {"Ĥ", "h"},
+        {"ĥ", "h"},  {"Ì", "i"},  {"ì", "i"},  {"Í", "i"},  {"í", "i"},  {"ı", "i"},  {"Ī", "i"},  {"ī", "i"},
+        {"Į", "i"},  {"į", "i"},  {"ï", "i"},  {"Ï", "i"},  {"î", "i"},  {"Î", "i"},  {"Ĵ", "j"},  {"ĵ", "j"},
+        {"Ķ", "k"},  {"ķ", "k"},  {"Ĺ", "l"},  {"ĺ", "l"},  {"Ļ", "l"},  {"ļ", "l"},  {"Ł", "l"},  {"ł", "l"},
+        {"Ľ", "l"},  {"ľ", "l"},  {"Ñ", "n"},  {"ñ", "n"},  {"Ń", "n"},  {"ń", "n"},  {"Ň", "n"},  {"ň", "n"},
+        {"Ņ", "n"},  {"ņ", "n"},  {"ô", "o"},  {"Ô", "o"},  {"ö", "o"},  {"Ö", "o"},  {"ò", "o"},  {"Ò", "o"},
+        {"Ó", "o"},  {"ó", "o"},  {"Õ", "o"},  {"õ", "o"},  {"Ő", "o"},  {"ő", "o"},  {"Ø", "o"},  {"ø", "o"},
+        {"Ŕ", "r"},  {"ŕ", "r"},  {"Ř", "r"},  {"ř", "r"},  {"ẞ", "ss"}, {"ß", "ss"}, {"Ś", "s"},  {"ś", "s"},
+        {"Ŝ", "s"},  {"ŝ", "s"},  {"Ş", "s"},  {"ş", "s"},  {"Š", "s"},  {"š", "s"},  {"Ș", "s"},  {"ș", "s"},
+        {"Ť", "t"},  {"ť", "t"},  {"Ţ", "t"},  {"ţ", "t"},  {"Ț", "t"},  {"ț", "t"},  {"Ú", "u"},  {"ú", "u"},
+        {"Ű", "u"},  {"ű", "u"},  {"Ũ", "u"},  {"ũ", "u"},  {"Ų", "u"},  {"ų", "u"},  {"Ů", "u"},  {"ů", "u"},
+        {"Ū", "u"},  {"ū", "u"},  {"û", "u"},  {"Û", "u"},  {"ù", "u"},  {"Ù", "u"},  {"ü", "u"},  {"Ü", "u"},
+        {"Ŵ", "w"},  {"ŵ", "w"},  {"Ý", "y"},  {"ý", "y"},  {"Ÿ", "y"},  {"ÿ", "y"},  {"Ŷ", "y"},  {"ŷ", "y"},
+        {"Ź", "z"},  {"ź", "z"},  {"Ž", "z"},  {"ž", "z"},  {"Ż", "z"},  {"ż", "z"},  {"Œ", "oe"}, {"œ", "oe"},
+        {"Æ", "ae"}, {"æ", "ae"}, {"Þ", "th"}, {"þ", "th"}, {"’", "'"}
+    };
+
+    for (const auto& vec : vec_str) {
+        boost::algorithm::replace_all(str, vec.first, vec.second);
+    }
+    return str;
+}
+
+std::string strip_accents_and_lower(const std::string& str) {
```

Comment:
Wouldn't a non const reference be better here to avoid a copy in `strip_accents` ?

Reply:
We do need a copy of the string because of `boost::algorithm::replace_all(...)` that mutates the string.